### ISSUE TITLE
Getting Value of parameter 'statuses' (TO_REVIEW) must be one of: [OPEN, CONFIRMED, REOPENED, RESOLVED,  error

### ISSUE THREAD

**Issue by praveendvd** [soprasteria/sonar-report#45]
![image](https://user-images.githubusercontent.com/45095911/97792700-76357480-1bd9-11eb-80ae-121199d89f27.png)

I am getting the below error : 

h status code 400:
{"errors":[{"msg":"Value of parameter 'statuses' (TO_REVIEW) must be one of: [OPEN, CONFIRMED, REOPENED, RESOLVED, CLOSED]"}]}

Command used:

```
sonar-report   \
--sonarurl="http://localhost:9000"   \
--sonarcomponent="Juiceshop"   \
--project="Juiceshop"   \
--application="Juiceshop"   \
--release="1.0.0"   \
--sinceleakperiod="false"  \
--sonartoken="ee622418df1026894573cc23f6d10e5319feffe1"   \
--allbugs="true" > /tmp/sonar-report_sonar-report.html

```

Sonarqube Version : Version 8.5.1 community

**Comment by praveendvd** [comment]
Setting , --noSecurityHotspot="true"  fixed this . I am not sure if this is the recommended fix

**Comment by ptrovatelli** [comment]
hello @praveendvd . 
That is strange. the security hotstpot was added in sonar 7.3; with it, the TO_REVIEW status was added. Are you certain that you run sonar 8.5? Can you double-check by browsing `<your_sonar_instance>/api/server/version` ? 

Can you post the content of the statuses supported in your instance? browse to `https://sonarqube-defectdojo-demo.apps.cicd.arcus.soprasteria.com/web_api/api/issues`, click  `parameter`, look for `statuses`
For example for sonar 7.9 I have: 
```
Possible values
OPEN
CONFIRMED
REOPENED
RESOLVED
CLOSED
TO_REVIEW
IN_REVIEW
REVIEWED
```

Maybe they removed the TO_REVIEW status in sonar 8, in which case we'll need to fix that in sonar-report.  

Using the `--noSecurityHotspot="true"` fixes the issue because sonar-report will act as if it's talking to a sonar 6 that doesn't support security hotspots vulnerabilities, but with a recent sonar, the problem is that you're going to miss all the security hotspots vulnerabilites

**Comment by christophe226** [comment]
I have the same problem on a sonar 8.2. When I look to statuses, possible values are:
- OPEN
- CONFIRMED
- REOPENED
- RESOLVED
- CLOSED

**Comment by cckoble** [comment]
I am on Sonar 8.3.1 and can verify the same results as other users with 8+

**Comment by ptrovatelli** [comment]
OK so it seems that the "TO_REVIEW" status was removed in sonar 8 while the security hotspot still exists. Or do you not have a security hotstpot either on your instances for some reason? In which case it would be safe to keep using the `--noSecurityHotspot="true"`

can you check the main project page in sonarqube, do you see "security hotspot" under the security section? 
![image](https://user-images.githubusercontent.com/34663482/106359295-105ca400-6312-11eb-892d-ccad442c99c8.png)

**Comment by christophe226** [comment]
Under /web_api/api/issues/search, types are:
```
CODE_SMELL
BUG
VULNERABILITY
```

And for what it worth, Project overview is displayed that way, not the same as yours:
![hotspots](https://user-images.githubusercontent.com/61734671/106735636-c971f580-6614-11eb-9a51-49fc783f6733.jpg)

**Comment by ptrovatelli** [comment]
@christophe226  ok great; so the security hotspot still exists but the issue type "SECURITY_HOTSTPOT" was removed. So the `--noSecurityHotspot="true"` can safely be used as it will extract all bugs, even those in the security hotspot. I'll update the documentation then: the flag should be used with sonar 6 and 8.

**Comment by christophe226** [comment]
Actually, in Sonar 8, `--noSecurityHotspot=true` does not extract security hotspots.
It may be an endless problem: we cannot call SECURITY_HOTSPOT type through API because this type was removed in Sonar 8 types list, but as we can only use types=VULNERABILITY in our API call, Sonar returns only vulnerabilities, not security hotspots.
It looks like a problem of Sonar having apparently internal issue types (vulnerabilities, bugs, code smells, security hotspots) which are not all reflected in their API issue types list (CODE_SMELL, BUG, VULNERABILITY).

**Comment by ptrovatelli** [comment]
ah, actually it seems they have moved the hotspots to another api endpoint. See this thread https://community.sonarsource.com/t/web-api-in-sonarqube-8-4-to-get-detailed-report-of-the-security-hotspots/31303 

They are supposed to be here starting with sonarqube 8: `/api/hotspots/search`

**Comment by reksii** [comment]
Hello,

I came across this issue recently when setting up this tool to generate reports. I get exactly the same error as in the title with a sonar 7.6.0
The possible issue statuses are OPEN, CONFIRMED, REOPENED, RESOLVED, CLOSED.
The possible issue types are CODE_SMELL, BUG, VULNERABILITY, SECURITY_HOTSPOT.

It works when I set noSecurityHotspot to true but it is a no go since I really need them in the report.
